content/tutorial-deep-learning-on-mnist.ipynb

```diff
@@ -0,0 +1,1017 @@
+{
```

Comment:
<p>Here, and below, you mention reshaping the data to be <code>28x28</code>.  Can you add a note explaining why that is necessary?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/numpy/numpy-tutorials/pull/33/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Reply:
That's a great point - done. Here is the diff:

```diff
3. Extract and save the data to disk using NumPy for the first time to read the contents of each file as [`ndarrays`](https://numpy.org/doc/stable/reference/arrays.ndarray.html) and saving them into a dictionary.
+ Note that you need to unroll (reshape) the original 28x28-pixel images into 1D vectors
+ (of size 28 times 28 or 784) — it is the expected shape of the input for your neural
+ network's first layer and the number of input neurons per image sample.
```

(without the 80-char limit)